client/client.go

```diff
@@ -108,7 +115,11 @@ var defaultRequiredServiceTypes = []string{"compute", "object-store"}
 
 func newClient(creds *identity.Credentials, auth_method identity.AuthMode, httpClient *goosehttp.Client, logger *log.Logger) AuthenticatingClient {
 	client_creds := *creds
-	client_creds.URL = client_creds.URL + apiTokens
+	if auth_method == identity.AuthUserPassV3 {
```

Comment:
in cases like this, using switch is better. it explicitly enumerates the auth_methods supported and allows an error to be returned in case a new one is added and the code now updated properly